cobbler/enums.py

```diff
@@ -1,17 +1,49 @@
 """
-TODO
+This module is responsible for containing all enums we use in Cobbler. It should not be dependent upon any other module
+except the Python standard library.
 """
 
 import enum
+from typing import Union
 
 VALUE_INHERITED = "<<inherit>>"
 VALUE_NONE = "none"
 
 
-class ResourceAction(enum.Enum):
+class ConvertableEnum(enum.Enum):
+    """
+    Abstract class to convert the enum via our convert method.
+    """
+
+    @classmethod
+    def to_enum(cls, value: Union[str, "ConvertableEnum"]) -> "ConvertableEnum":
```

Comment:
I still don't fully understand the meta programming that makes e.g. `__getattr__` seemingly impossible. 

I came up with a very similar solution, the main difference is actually the method name. To me, `to_enum` suggests that a new enum is constructed and not that an existing enum is looked up. The type of the returned object _is_ (a subclass of) `enum.Enum`, but I normally come across `to_foo` as a method that returns a new representation of itself using another type. Actually I most often see `to_something` in Rust code, in Python operations like that can also be done using "magic methods".

Just for the sake of sharing what I have:
```python
class MyEnum(enum.Enum):
    @classmethod
    def item(cls, str_or_enum):
        try:
            if isinstance(str_or_enum, str):
                return cls[str_or_enum.upper()]
            elif isinstance(str_or_enum, cls):
                return str_or_enum
            else:
                raise TypeError(f"{str_or_enum} must be a str or Enum")
        except KeyError:
            raise ValueError(f"{str_or_enum} must be one of {list(cls)}")
```

Reply:
I find your method more robust then mine and a littlebit easier to read. Thus I will incorporate your suggestion into the code.